datastore/longhorn.go

```diff
@@ -965,13 +965,27 @@ func (s *DataStore) CreateDefaultNode(name string) (*longhorn.Node, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	kubeNode, err := s.GetKubernetesNode(name)
+	if err != nil {
+		return nil, err
+	}
+
+	isNewVersion, err := util.IsKubernetesVersionAtLeast(s.kubeClient, types.KubernetesNewZoneLabelVersion)
+	if err != nil {
+		return nil, err
+	}
+	region, zone := types.GetRegionAndZone(kubeNode.Labels, isNewVersion)
+
 	node := &longhorn.Node{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: name,
 		},
 		Spec: types.NodeSpec{
 			Name:            name,
 			AllowScheduling: true,
+			Region:          region,
```

Comment:
Probably should be in the node status instead of the spec? It's reflecting what's in the Kubernetes node, not configurable from the Longhorn part.